java/daikon/dcomp/BuildJDK.java

```diff
@@ -90,288 +57,380 @@ protected boolean exists(String libRt) {
-   *
-   * <dl>
-   *   <dt>jarfile
-   *   <dd>jarfile to process
-   *   <dt>dest
-   *   <dd>destination directory in which to place instrumented classes
-   *   <dt>prefix
-   *   <dd>optional prefix of classes to be translated
-   * </dl>
-   *
-   * Instruments each class file in jarfile that begins with prefix and puts the results in dest.
-   */
+  // We want to share code to read members of a jar file (JDK 8) or members of a
+  // module file (JDK 9+). We accomplish this by saving an InputStream for
+  // each member class we find.
+  private Map<String, InputStream> class_stream_map = new HashMap<>();
+
+  /** Instruments each class file in Java runtime and puts the result in dest. */
   public static void main(String[] args) throws IOException {
 
-    System.out.println("Starting at " + new Date());
+    System.out.println("BuildJDK starting at " + new Date());
 
-    Options options = new Options(synopsis, BuildJDK.class, DynComp.class);
-    // options.ignore_options_after_arg (true);
+    Options options =
+        new Options(
+            "daikon.BuildJDK [options] dest [classfiles...]", BuildJDK.class, DynComp.class);
     String[] cl_args = options.parse(true, args);
-    boolean ok = check_args(options, cl_args);
-    if (!ok) System.exit(1);
+    check_args(options, cl_args);
     verbose = DynComp.verbose;
 
+    BuildJDK build = new BuildJDK();
+    File dest_dir = new File(cl_args[0]);
+
     if (classfiles) {
 
-      // Arguments are <srcdir> <destdir> <classfiles>...
-      File src_dir = new File(cl_args[0]);
-      File dest_dir = new File(cl_args[1]);
-      File[] class_files = new File[cl_args.length - 2];
-      for (int ii = 2; ii < cl_args.length; ii++) {
-        class_files[ii - 2] = new File(cl_args[ii]);
+      // Arguments are <destdir> <classfiles>...
+      File[] class_files = new File[cl_args.length - 1];
+      for (int i = 1; i < cl_args.length; i++) {
+        class_files[i - 1] = new File(cl_args[i]);
       }
 
-      BuildJDK build = new BuildJDK();
-
-      // Restore the static map from field names to ids
+      // The classfiles argument is usually used for testing.
+      // But if we're using it to fix a broken classfile, then we need
+      // to restore the static map from when our runtime jar was originally
+      // built.  We assume it is in the destination directory.
       DCInstrument.restore_static_map(new File(dest_dir, static_map_fname));
-      System.out.printf("Restored %d entries in static map%n", DCInstrument.static_map.size());
+      System.out.printf("Restored %d entries in static map.%n", DCInstrument.static_map.size());
 
       // Read in each specified classfile
-      Map<String, JavaClass> classmap = new LinkedHashMap<>();
       for (File class_file : class_files) {
-        if (class_file.toString().endsWith("java/lang/Object.class")) {
-          System.out.printf("Skipping %s%n", class_file);
+        String className = class_file.toString();
```

Comment:
This variable named `className` does not hold a class name.  It holds a file name.